main.c

```diff
@@ -251,6 +280,7 @@ MPZ_from_str(PyObject *s, int base)
 
     Py_ssize_t len;
     const char *str = PyUnicode_AsUTF8AndSize(s, &len);
+    underscores_as_number((char *)str, &len);
```

Comment:
Put newline before and remove newline after.

Note also, that you introduce memory leak on failure.